cmd/build/v2/services_test.go

```diff
@@ -186,15 +196,111 @@ func TestGetToBuildTag(t *testing.T) {
 				Context:    ".",
 				Image:      "okteto.dev/test:test",
 			},
+			isOkteto:     true,
+			manifestName: "test",
+			svcName:      "test",
+			output:       []string{"okteto.dev/test:test"},
+		},
+		{
+			name: "image is set without volume mounts",
+			buildInfo: &model.BuildInfo{
+				Dockerfile: "Dockerfile",
+				Context:    ".",
+				Image:      "test/test:test",
+			},
+			isOkteto:     true,
 			manifestName: "test",
 			svcName:      "test",
-			output:       "okteto.dev/test:test",
+			output: []string{
+				"okteto.dev/test-test:okteto",
+				"okteto.global/test-test:okteto",
+				"test/test:test",
+			},
 		},
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			result := getToBuildTag(tt.manifestName, tt.svcName, tt.buildInfo)
+			okteto.CurrentStore = &okteto.OktetoContextStore{
+				Contexts: map[string]*okteto.OktetoContext{
+					"test": {
+						Namespace: "test",
+						IsOkteto:  tt.isOkteto,
+						Registry:  "https://registry.test",
+					},
+				},
+				CurrentContext: "test",
+			}
+			result := getToBuildTags(tt.manifestName, tt.svcName, tt.buildInfo)
 			assert.Equal(t, tt.output, result)
 		})
 	}
 }
+
+func TestGetDigestFromService(t *testing.T) {
+	fakeReg := test.NewFakeOktetoRegistry(nil)
+
+	tests := []struct {
+		name           string
+		manifest       *model.Manifest
+		isOkteto       bool
+		expectedErr    bool
+		expectedDigest bool
+	}{
+		{
+			name: "image is set not okteto cluster",
+			manifest: &model.Manifest{
+				Build: model.ManifestBuild{
+					"test": {
+						Dockerfile: "Dockerfile",
+						Context:    ".",
+					},
+				},
+			},
+			isOkteto:       false,
+			expectedErr:    true,
+			expectedDigest: false,
+		},
+		{
+			name: "image not in registry",
+			manifest: &model.Manifest{
+				Build: model.ManifestBuild{
+					"test": {
+						Dockerfile: "Dockerfile",
+						Context:    ".",
+					},
+				},
+			},
+			isOkteto:       true,
+			expectedErr:    false,
+			expectedDigest: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			okteto.CurrentStore = &okteto.OktetoContextStore{
+				Contexts: map[string]*okteto.OktetoContext{
+					"test": {
+						Namespace: "test",
+						IsOkteto:  tt.isOkteto,
+						Registry:  "https://registry.test",
+					},
+				},
+				CurrentContext: "test",
+			}
+
+			bc := NewBuilder(nil, fakeReg)
```

Comment:
```suggestion
			bc := NewBuilder(nil, fakeReg)
```
we can move this line to the top to avoid declare same var `bc` with same values inside the test loop